src/server/StudioManagementSystem/Controllers/StudentContactController.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using StudioManagementSystem.Core.Dtos;
+using StudioManagementSystem.Infrastructure.Interfaces.Data;
+
+namespace StudioManagementSystem.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class StudentContactController : ControllerBase
+{
+    private readonly IStudentContactRepository _studentContactRepository;
+    private readonly ICancellationTokenAccessor _cancellationTokenAccessor;
+
+    public StudentContactController(IStudentContactRepository studentContactRepository, ICancellationTokenAccessor cancellationTokenAccessor)
+    {
+        _studentContactRepository = studentContactRepository;
+        _cancellationTokenAccessor = cancellationTokenAccessor;
+    }
+    
+    [HttpPost]
+    public ActionResult<Guid> CreateStudentContact(CreateStudentContactDto dto)
+    {
+        var ct = _cancellationTokenAccessor.Token;
+        var task = _studentContactRepository.AddStudentContactAsync(new(dto), ct);
+        task.Wait(ct);
+
+        return task.Result;
+    }
+    
+    [HttpPost("[action]/{id}")]
+    [ActionName("leavegroup")]
+    public ActionResult<Guid> LeaveAssignedGroup(Guid id)
```

Comment:
Leaving a group is a boolean event ("did I or didn't I?"), returning an ID here is not needed. Also labelling the param as a group ID is meaningful here, as the controller is for a different entity.


```suggestion
    public ActionResult<bool> LeaveAssignedGroup(Guid groupId)
```